src/Exception/HostedDomainException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace League\OAuth2\Client\Exception;
+
+/**
+ * Exception thrown if the Google Provider is configured with a hosted domain that the user doesn't belong to
+ */
+class HostedDomainException extends \Exception
+{
+    private $hostedDomainConfigured;
+
+    private $hostedDomainOfUser;
+
+    /**
+     * HostedDomainException constructor.
+     * @param string $hostedDomainConfigured
+     * @param string|null $hostedDomainOfUser
+     */
+    public function __construct($hostedDomainConfigured, $hostedDomainOfUser)
+    {
+        parent::__construct("Hosted domain mismatch '$hostedDomainOfUser' !== '$hostedDomainConfigured'");
```

Comment:
This could be used as an XSS attack vector if the site has exception handling output. I think it would be better to only show the `$hostedDomainConfigured` value, as that will be developer supplied.